core/src/price_estimation/orderbook_based.rs

```diff
@@ -26,35 +32,19 @@ impl PriceSource for PricegraphEstimator {
             let pricegraph = Pricegraph::new(orders.iter().map(|order| {
                 order.to_element(account_state.read_balance(order.sell_token, order.account_id))
             }));
-            Ok(self.estimate_prices(tokens, pricegraph))
+            pricegraph.get_prices(tokens).await
```

Comment:
Is `impl PriceSource for PricegraphEstimator` still needed? As far as I understand we have `impl LimitPriceEstimating for Pricegraph` and `impl<T: LimitPriceEstimating> PriceSource for T` so shouldn't this work without this specific impl?

Reply:
`PriceSource` is implemented for all `LimitPriceEstimating` but `PricegraphEstimator` is not `LimitPriceEstimating`, only `Pricegraph` is `LimitPriceEstimating`. Indeed when would the code from this function that creates the orderbook be executed if we didn't have this implementation.

Reply:
I see (so many structs with very similar names). I guess we could also `impl LimitPriceEstimating for PricegraphEstimator` but that would create the orderbook n times (once for each token). Makes sense the way it is...